### PR TITLE
test: ensure that python ci/datamgr.py hdf5 produces data

### LINKED ISSUE #2998: TST: hdf5 tests are skipped
It looks like most or all of the backend tests for the `hdf5` backend are skipped, because we're not creating any of the necessary testing data.

PR incoming.

### PR DESCRIPTION
**Pull request by cpcloud**
This PR adds data generation for the hdf5 backend.

Closes #2998

### CODE REVIEW

**Review by jreback — APPROVED** [review]
lgtm merge on green

**Comment by cpcloud** [comment]
@jreback @datapythonista Can one of y'all give this another review? See my previous comment, I had to make a few changes beyond just flipping on the data.

File: ibis/backends/hdf5/__init__.py [hunk 1]
```diff
@@ -23,19 +23,25 @@ def insert(
...
         with pd.HDFStore(str(path), mode='r') as store:
-            df = store.select(name, start=0, stop=0)
+            df = store.select(name, start=0, stop=50)
             schema = sch.infer(df)
 
         t = self.table_class(name, schema, self).to_expr()
         self.dictionary[name] = path
         return t
 
+    def compile(self, expr, *args, **kwargs):
```
Comment (jreback) [inline ibis/backends/hdf5/__init__.py:42]: odd that this isn't in the base class

**Review by jreback — APPROVED** [review]
looks fine to me

**Review by datapythonista — COMMENTED** [review]
lgtm, but agree with Jeff, `compile` should be moved to the base file backend (and removed from csv and parquet backends), instead of adding it here.

**Comment by cpcloud** [comment]
Roger, moving it to the base class.

File: ibis/backends/hdf5/__init__.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/hdf5/__init__.py]: Done.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @cpcloud

**Pull request merged**